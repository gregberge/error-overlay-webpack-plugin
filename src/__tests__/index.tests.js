import {tapEntry} from "../index"

test("tapEntry should throw when receiving a string", () => {
  expect(() => tapEntry("/tmp", "main")).toThrowErrorMatchingSnapshot()
})

test("tapEntry should add entry to array", () => {
  const entry = ["main"]

  tapEntry("/tmp", entry)

  expect(entry[0]).toContain("entry.js")
  expect(entry[1]).toEqual("main")
})

test("tapEntry should add entry to arrays in object", () => {
  const entry = {main: ["main"], login: ["login"]}

  tapEntry("/tmp", entry)

  expect(entry.main[0]).toContain("entry.js")
  expect(entry.main[1]).toEqual("main")
  expect(entry.login[0]).toContain("entry.js")
  expect(entry.login[1]).toEqual("login")
})
