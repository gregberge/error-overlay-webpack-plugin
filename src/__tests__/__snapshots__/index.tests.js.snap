// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tapEntry should throw when receiving a string 1`] = `"error-overlay-webpack-plugin does not support string entry"`;
